native/cocos/platform/BasePlatform.cpp

```diff
@@ -43,10 +43,18 @@
 #endif
 
 namespace cc {
-BasePlatform::BasePlatform()  = default;
-BasePlatform::~BasePlatform() = default;
+BasePlatform* BasePlatform::_currentPlatform = nullptr;
 
-BasePlatform* BasePlatform::getPlatform() {
+BasePlatform::BasePlatform() {
+    CCASSERT(_currentPlatform == nullptr, "The platform has been initialized, only one platform can be initialized");
+    _currentPlatform = this;
+}
+
+BasePlatform::~BasePlatform() {
+    _currentPlatform = nullptr;
+}
+
+BasePlatform* BasePlatform::createDefaultPlatform() {
```

Comment:
Yes here, but better is to assign to currentPlatform. But because of the assignment in the constructor (to avoid letting the user set it). It feels hard to understand here. In fact, is it okay for me to assign a value here, although there is redundancy？

Reply:
This will not be modified, it will be strange if the inside of this function is modified. Let's see how this place is handled